src/WebJobs.Script.WebHost/Global.asax.cs

```diff
@@ -45,6 +47,25 @@ private static void VerifyAndEnableShadowCopy(WebHostSettings webHostSettings)
             }
         }
 
+        private static void ConfigureMinimumThreads(bool isDynamicSku)
+        {
+            // For information on MinThreads, see:
+            // https://docs.microsoft.com/en-us/dotnet/api/system.threading.threadpool.setminthreads?view=netframework-4.7
+            // https://docs.microsoft.com/en-us/azure/redis-cache/cache-faq#important-details-about-threadpool-growth
+            // https://blogs.msdn.microsoft.com/perfworld/2010/01/13/how-can-i-improve-the-performance-of-asp-net-by-adjusting-the-clr-thread-throttling-properties/
+            //
+            // This behavior can be overridden by using the "ComPlus_ThreadPool_ForceMinWorkerThreads" environment variable (honored by the .NET threadpool).
+
+            // The dynamic plan has some limits that mean that a given instance is using effectively a single core, so we should not use Environment.Processor count in this case.
+            var effectiveCores = isDynamicSku ? 1 : Environment.ProcessorCount;
```

Comment:
Its necessary. Here's some code I just ran on a consumption function app (without this fix), and its output:

```csharp
public static void Run(HttpRequestMessage req, TraceWriter log)
{
    log.Info(Environment.ProcessorCount.ToString());
    log.Info("C# HTTP trigger function processed a request.");

    System.Threading.ThreadPool.GetMinThreads(out int x, out int y);
    log.Info($"{x},{y}");
}
```

```
2017-07-19T15:53:07.150 Function started (Id=c3797636-6f5a-4ce4-abd7-eb43b38f9c3c)
2017-07-19T15:53:07.150 2
2017-07-19T15:53:07.150 C# HTTP trigger function processed a request.
2017-07-19T15:53:07.150 1,1
2017-07-19T15:53:07.150 Function completed (Success, Id=c3797636-6f5a-4ce4-abd7-eb43b38f9c3c, Duration=3ms)
```

I checked the logs and this code happened to execute on a medium VM.